### ISSUE TITLE
MinMax does not operate well when value is small

### ISSUE THREAD

**Issue by dingjr27** [stillwater-sc/universal#270]
Problem:

In include/universal/number/posit/math/minmax.hpp,
expression sw::universal::posit<32,2>(std::pow(double(0.00978569), double(std::numeric_limits<sw::universal::posit<32,2>>::min())))
gives 1 rather than 0.00978569.

**Comment by Ravenwater** [comment]
std::numeric_limits<posit<32, 2>>::min() represents minpos of the posit<32,2> and is around 7.52316e-37. 

the value of 0.00978569^7.52316e-37 = 1

Why do you think pow(x,y) with the above values would be x?

**Comment by Ravenwater** [comment]
![image](https://user-images.githubusercontent.com/466641/149415960-47a4d096-e3ae-466a-967a-a49dd0e4be8d.png)

**Comment by dingjr27** [comment]
Dear developer,

I did not argue that 0.00978569^7.52316e-37 = 1. What i want to express is that max(0.00978569,7.52316e-37) = 1 is wrong.

I discovered the bug during the usage of Eigen. Its call routine is like numext::maxi(A,B) -> std::max(A,B) -> sw::universal::max(A,B). I originally thought this expression is optimized for speed so it looks strange. So i thought the problem was caused by one of operator being too small.

In "minmax.hpp"
```
template<size_t nbits, size_t es>
posit<nbits,es> min(posit<nbits,es> x, posit<nbits, es> y) {
	return posit<nbits,es>(std::pow(double(x), double(y)));
}

template<size_t nbits, size_t es>
posit<nbits,es> max(posit<nbits,es> x, posit<nbits, es> y) {
	return posit<nbits,es>(std::pow(double(x), double(y)));
}
```
Best regards!

**Comment by Ravenwater** [comment]
Sorry, I totally misunderstood your observation. That was definitely a copy and paste error. I have checked-in a fix, and next release (v3.51) will have this fixed. I'll release it before the end of the week.

FYI: on the todo list is to create native posit implementations using the Minefield method, but the Minefield polynomials are different for each posit configuration, so we haven't figured out yet how to approach that problem for Universal. The simple approach exhibits double rounding failures, so we are not yet correctly rounding across the full domain.

**Comment by Ravenwater** [comment]
fixed in v3.51 and merged in main
